diff --git a/cmake/MirCommon.cmake b/cmake/MirCommon.cmake
index d6149be..92d04eb 100644
--- a/cmake/MirCommon.cmake
+++ b/cmake/MirCommon.cmake
@@ -319,6 +319,12 @@ function (mir_generate_protocol_wrapper TARGET_NAME NAME_PREFIX PROTOCOL_FILE)
   if (NAME_PREFIX STREQUAL "")
     set(NAME_PREFIX "@") # won't match anything
   endif()
+  if (NOT DEFINED MIR_WAYLAND_GENERATOR_EXECUTABLE)
+    set(MIR_WAYLAND_GENERATOR_EXECUTABLE "${CMAKE_BINARY_DIR}/bin/mir_wayland_generator")
+    set(MIR_WAYLAND_GENERATOR_DEPENDS mir_wayland_generator)
+  else()
+    set(MIR_WAYLAND_GENERATOR_DEPENDS "")
+  endif()
   get_filename_component(PROTOCOL_NAME "${PROTOCOL_FILE}" NAME_WE)
   set(OUTPUT_PATH_HEADER "${CMAKE_CURRENT_BINARY_DIR}/${PROTOCOL_NAME}_wrapper.h")
   set(OUTPUT_PATH_SRC "${CMAKE_CURRENT_BINARY_DIR}/${PROTOCOL_NAME}_wrapper.cpp")
@@ -327,10 +333,10 @@ function (mir_generate_protocol_wrapper TARGET_NAME NAME_PREFIX PROTOCOL_FILE)
     OUTPUT "${OUTPUT_PATH_HEADER}" "${OUTPUT_PATH_SRC}"
     VERBATIM
     COMMAND "sh" "-c"
-    "${CMAKE_BINARY_DIR}/bin/mir_wayland_generator ${NAME_PREFIX} ${PROTOCOL_PATH} header > ${OUTPUT_PATH_HEADER}"
+    "${MIR_WAYLAND_GENERATOR_EXECUTABLE} ${NAME_PREFIX} ${PROTOCOL_PATH} header > ${OUTPUT_PATH_HEADER}"
     COMMAND "sh" "-c"
-    "${CMAKE_BINARY_DIR}/bin/mir_wayland_generator ${NAME_PREFIX} ${PROTOCOL_PATH} source > ${OUTPUT_PATH_SRC}"
-    DEPENDS mir_wayland_generator "${PROTOCOL_PATH}"
+    "${MIR_WAYLAND_GENERATOR_EXECUTABLE} ${NAME_PREFIX} ${PROTOCOL_PATH} source > ${OUTPUT_PATH_SRC}"
+    DEPENDS ${MIR_WAYLAND_GENERATOR_DEPENDS} "${PROTOCOL_PATH}"
   )
   target_sources("${TARGET_NAME}" PRIVATE "${OUTPUT_PATH_HEADER}" "${OUTPUT_PATH_SRC}")
 endfunction()
