diff --git a/CMakeLists.txt b/CMakeLists.txt
index e014eee..8c9dbdf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -300,7 +300,10 @@ set(MIR_GENERATED_INCLUDE_DIRECTORIES)
 add_subdirectory(src/)
 include_directories(${MIR_GENERATED_INCLUDE_DIRECTORIES})
 
-add_subdirectory(examples/)
+option(MIR_ENABLE_EXAMPLES "Build examples" ON)
+if (MIR_ENABLE_EXAMPLES)
+  add_subdirectory(examples/)
+endif()
 add_subdirectory(guides/)
 add_subdirectory(cmake/)
 
